### PR TITLE
SQL database

### PR DESCRIPTION
**Pull request by kawamet**
#5

### CODE REVIEW

File: src/main/java/uk/wanat/theclinick/model/Doctor.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public Doctor(Speciality speciality, String firstName, String lastName) {
+        this.speciality = speciality;
+        this.firstName = firstName;
+        this.lastName = lastName;
+    }
+
+    public void add(Appointment tempAppointment) {
+
+        if (appointments == null) {
+            appointments = new ArrayList<>();
```
Comment (mmm83) [inline src/main/java/uk/wanat/theclinick/model/Doctor.java]: Lazy inits can be prone to race conditions. I'd initialize it in constructor.

File: src/main/java/uk/wanat/theclinick/controller/PatientController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                model.addAttribute("patient", patientTemp);
+                return "patient/patient-already-exists";
+            }
+        }
+        patientServiceImpl.create(patient);
+        return "patient/patient-created";
+    }
+
+    @RequestMapping("patient/{id}/update")
+    public String updateUpdate(@PathVariable String id, Model model) {
```
Comment (mmm83) [inline src/main/java/uk/wanat/theclinick/controller/PatientController.java]: Is this method name what you intended?

File: src/main/java/uk/wanat/theclinick/controller/PatientController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public String addPatient(@ModelAttribute Patient patient, Model model) {
+        List<Patient> all = patientServiceImpl.findAll();
+        for (Patient patientTemp : all) {
+            if (patientTemp.getNationalInsuranceNumber().equals(patient.getNationalInsuranceNumber())) {
+                model.addAttribute("patient", patientTemp);
+                return "patient/patient-already-exists";
+            }
+        }
+        patientServiceImpl.create(patient);
+        return "patient/patient-created";
```
Comment (mmm83) [inline src/main/java/uk/wanat/theclinick/controller/PatientController.java]: General comment for controllers. Looks like there is some room for unittesting in all controller classes. Consider adding tests for those classes, even if just for practice.

File: src/main/java/uk/wanat/theclinick/controller/PatientController.java (hunk 2 above)
Comment (kawamet) [reply src/main/java/uk/wanat/theclinick/controller/PatientController.java]: It was mistakenly written. It should be as below:
1. 
```suggestion
    public String updatePatient(@PathVariable String id, Model model) {
```

File: src/main/java/uk/wanat/theclinick/model/Doctor.java (hunk 1 above)
Comment (kawamet) [reply src/main/java/uk/wanat/theclinick/model/Doctor.java]:
```suggestion

       public Doctor(Speciality speciality, String firstName, String lastName) {
        this.speciality = speciality;
        this.firstName = firstName;
        this.lastName = lastName;
        appointments = new ArrayList<>();
    }
```
```suggestion

        public void add(Appointment tempAppointment) {
        appointments.add(tempAppointment);
        tempAppointment.setDoctor(this);
    }
```

Comment (kawamet) [reply src/main/java/uk/wanat/theclinick/model/Doctor.java]:
```suggestion
           
      
              appointments = new ArrayList<>();
```

Comment (kawamet) [reply src/main/java/uk/wanat/theclinick/model/Doctor.java]:
```suggestion
        public Doctor(Speciality speciality, String firstName, String lastName) {
        this.speciality = speciality;
        this.firstName = firstName;
        this.lastName = lastName;
        appointments = new ArrayList<>();
    }
```

File: src/main/java/uk/wanat/theclinick/controller/PatientController.java (hunk 3 above)
Comment (kawamet) [reply src/main/java/uk/wanat/theclinick/controller/PatientController.java]: Sure. I added a new task: #8. 
I will focus on it in the next part of the project.

File: src/main/java/uk/wanat/theclinick/model/Doctor.java (hunk 1 above)
Comment (kawamet) [reply src/main/java/uk/wanat/theclinick/model/Doctor.java]: Changes made in commit 9a2c775 and b628080. ArrayList initialized in constructor

**Pull request merged**